parser/src/token.rs

```diff
@@ -0,0 +1,86 @@
+use std::fmt;
+pub enum Tok {
+    Import,
+    Declare,
+    BoundaryConstraints,
+    TraceColumnns,
+    TransitionConstraints,
+    Main,
+    PeriodicColumns,
+    Assert,
+    Aux,
+    State,
+    Enforce,
+    Ev,
+    Let,
+    With,
+    First,
+    Last,
+    For,
+    In,
+    PubInputs,
+    ProgramHash,
+    StackInputs,
+    StackOutputs,
+    Rand,
+    Sum,
+    Colon,
+    Comma,
+    Eqequal,
+    Equal,
+    Lpar,
+    Rpar,
+    Lsqb,
+    Rsqb,
+    Star,
+    Minus,
+    MinusEqual,
+    Plus,
+    PlusEqual,
+    Rarrow
+}
+
+impl fmt::Display for Tok {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        match self {
+            Self::Import => write!(f, "import"),
+            Self::Declare => write!(f, "declare"),
+            Self::BoundaryConstraints => write!(f, "boundary_constraints"),
+            Self::TraceColumnns => write!(f, "trace_columns"),
+            Self::TransitionConstraints => write!(f, "transition_constraints"),
+            Self::Main => write!(f, "main"),
+            Self::PeriodicColumns => write!(f, "periodic_columns"),
+            Self::Assert => write!(f, "assert"),
+            Self::Aux => write!(f, "aux"),
+            Self::State => write!(f, "state"),
+            Self::Enforce => write!(f, "enf"),
+            Self::Ev => write!(f, "ev"),
+            Self::Let => write!(f, "let"),
+            Self::With => write!(f, "with"),
+            Self::First => write!(f, "first"),
+            Self::Last => write!(f, "last"),
+            Self::For => write!(f, "for"),
+            Self::In => write!(f, "in"),
+            Self::PubInputs => write!(f, "pub_inputs"),
+            Self::ProgramHash => write!(f, "program_hash"),
+            Self::StackInputs => write!(f, "stack_inputs"),
+            Self::StackOutputs => write!(f, "stack_outputs"),
+            Self::Rand => write!(f, "rand"),
+            Self::Sum => write!(f, "sum"),
+            Self::Colon => write!(f, ":"),
+            Self::Comma => write!(f, ","),
+            Self::Eqequal => write!(f, "=="),
+            Self::Equal => write!(f, "="),
+            Self::Lpar => write!(f, "("),
+            Self::Rpar => write!(f, ")"),
+            Self::Lsqb => write!(f, "["),
+            Self::Rsqb => write!(f, "]"),
+            Self::Star => write!(f, "*"),
+            Self::Minus => write!(f, "-"),
+            Self::MinusEqual => write!(f, "-="),
+            Self::Plus => write!(f, "+"),
+            Self::PlusEqual => write!(f, "+="),
+            Self::Rarrow => write!(f, "->"),
```

Comment:
This is just a single comment (I haven't really looked at other parts much): We should probably start with the smallest set of keywords we need for now, and then expand this in future PRs. So, things like `import`, `for`, `in` etc. should probably be removed from here for now.

Reply:
I've removed all keywords that we don't use in the current test file.